content/docs/iac/extending-pulumi/use-terraform-module.md

```diff
@@ -204,13 +208,544 @@ function getCidrSubnet(cidr: string, netnum: number): pulumi.Output<string> {
+
+When authoring in YAML, there's no need for Pulumi to generate a SDK. In the YAML you can reference the Terraform module by its schema token, which takes the format `<module-name>:index:Module`:
+
+**Example:** `Pulumi.yaml` - Using the Terraform VPC and RDS module in a Pulumi program
+
+```yaml
+name: testproj-yaml
+description: testproj-yaml
+runtime: yaml
+resources:
+  testVpc:
+    type: vpcmod:index:Module
+    properties:
+      name: test-vpc-${pulumi.stack}
+      azs:
+        - us-west-2a
+        - us-west-2b
+        - us-west-2c
+      cidr: 10.0.0.0/16
+      public_subnets:
+        - 10.0.1.0/24
+        - 10.0.2.0/24
+        - 10.0.3.0/24
+      private_subnets:
+        - 10.0.5.0/24
+        - 10.0.6.0/24
+        - 10.0.7.0/24
+      database_subnets:
+        - 10.0.9.0/24
+        - 10.0.10.0/24
+        - 10.0.11.0/24
+      create_database_subnet_group: true
+  testRdsSg:
+    type: aws:ec2:SecurityGroup
+    properties:
+      vpcId: ${testVpc.vpc_id}
+  testRdsSgIngress:
+    type: aws:vpc:SecurityGroupIngressRule
+    properties:
+      ipProtocol: tcp
+      securityGroupId: ${testRdsSg.id}
+      cidrIpv4: ${testVpc.vpc_cidr_block}
+      fromPort: 3306
+      toPort: 3306
+  testRds:
+    type: rdsmod:index:Module
+    properties:
+      engine: mysql
+      identifier: test-rds-${pulumi.stack}
+      manage_master_user_password: true
+      publicly_accessible: false
+      allocated_storage: 20
+      max_allocated_storage: 100
+      instance_class: db.t4g.large
+      engine_version: 8
+      family: mysql8.0
+      db_name: completeMysql
+      username: complete_mysql
+      port: '3306'
+      multi_az: true
+      db_subnet_group_name: ${testVpc.database_subnet_group_name}
+      vpc_security_group_ids:
+        - ${testRdsSg.id}
+      skip_final_snapshot: true
+      deletion_protection: false
+      create_db_option_group: false
+      create_db_parameter_group: false
+packages:
+  rdsmod:
+    source: terraform-module
+    version: 0.1.7
+    parameters:
+      - terraform-aws-modules/rds/aws
+      - 6.10.0
+      - rdsmod
+  vpcmod:
+    source: terraform-module
+    version: 0.1.7
+    parameters:
+      - terraform-aws-modules/vpc/aws
+      - 5.19.0
+      - vpcmod
+```
+
+{{% /choosable %}}
+
+{{% /chooser %}}
+
 In the above code, the imported Terraform module works the same as any other Pulumi code. Outputs are returned, and resource state is stored in your Pulumi state storage, alongside all your other Pulumi-native resources. This also means that resource dependencies work as expected between Pulumi-native resources and resources created by Terraform modules.
 
 ## Configuring Terraform Providers
 
 Some modules require Terraform providers to be configured with specific settings. You can configure these providers from within Pulumi:
 
-**Example:** index.ts - Configuring the imported Terraform bucket module*
+{{% chooser language "typescript,python,go,csharp,java" %}}
```

Comment:
Got a problem formulating this with YAML:

https://github.com/pulumi/pulumi-yaml/issues/824